core/domain/html_cleaner.py

```diff
@@ -531,6 +531,7 @@ def convert_to_ckeditor(html_data):
                 if prev_sib and prev_sib.name == 'li':
                     if 'style' in tag.attrs:
                         del tag['style']
+                    prev_sib.append(tag)
```

Comment:
Hang on -- this seems like the wrong approach to me. I don't understand why the RTE content needs to have style tags.

Shouldn't we just do something like: `<ul><li>...</li><li><ul><li>...</li></ul></li></ul>`? I just tried creating a nested list in CKEditor and it has a nested ul/li structure. Using margins to "visually" replicate the structure is hacky and I would discourage it.

Reply:
I'm not sure I follow. Try creating a nested list in CKEditor. Is there a styling applied to the HTML? From what I can tell, a nested list along the lines of 

- Item 1
  - Item 2

has HTML code as follows:

`<ul><li>Item 1<ul><li>Item 2</li></ul></li></ul>`

without the need for styling.

Reply:
Yes, right!
For the above case there would be no styling in the approach that I'm using (It will be the same output as you said). Let me give you another example where styling is necessary. 
```
<ol>
  <li>Nitish</li>
  <ol>
    <ol>
      <ol>
        <li>Ipsum</li>
        <li>Lorel</li>
     </ol>
   </ol>
  </ol>
</ol>
```
Here there is a list element `Nitish` and there is an indentation to create a nested list (created by using 3 times the indent button). For cases like these the approach you said won't work. To handle these cases where the nested list is indented by more than 1 tab/indent styling is necessary.

So when there are multiple nested `ol` tags one after the other there we need styling such that the list looks visually same in both, hence a styling of margin-left: 80px; will be added here since the nested list has a `ol` tag for nesting it and other two `ol` tags for indentation.

Reply:
I don't see the point of having a three-times nested list. Can we just get rid of those and bring them back to one level of nesting?

I.e.,

```
<ol>
  <li>Nitish</li>
  <ol>
    <ol>
      <ol>
        <li>Ipsum</li>
        <li>Lorel</li>
     </ol>
   </ol>
  </ol>
</ol>
```

produces

<ol>
  <li>Nitish</li>
  <ol>
    <ol>
      <ol>
        <li>Ipsum</li>
        <li>Lorel</li>
     </ol>
   </ol>
  </ol>
</ol>

and we should change it to

```
<ol>
  <li>
    Nitish
    <ol>
      <li>Ipsum</li>
      <li>Lorel</li>
    </ol>
  </li>
</ol>
```

which produces

<ol>
  <li>
    Nitish
    <ol>
      <li>Ipsum</li>
      <li>Lorel</li>
    </ol>
  </li>
</ol>

Reply:
Okay fine as you say!

Had I discussed this first it should have reduced a lot of my work. I spent much time testing and figuring this out (during my first PR too) :sweat_smile:.  I once thought what you said but then thought that lists should look similar so went that way.

No worries, will issue a PR and fix this, this is easy.

Thanks a lot @seanlip